From efeb651a86570eae98d98c7d8a57560caf51d55c Mon Sep 17 00:00:00 2001
From: Naushir Patuck <naush@raspberrypi.com>
Date: Fri, 31 Mar 2023 10:07:26 +0100
Subject: [PATCH] drivers: media: imx708: Remove unused control fields

Remove unused and redundant control fields from the state structure.

Signed-off-by: Naushir Patuck <naush@raspberrypi.com>
---
 drivers/media/i2c/imx708.c | 10 +++-------
 1 file changed, 3 insertions(+), 7 deletions(-)

--- a/drivers/media/i2c/imx708.c
+++ b/drivers/media/i2c/imx708.c
@@ -825,9 +825,6 @@ struct imx708 {
 	struct v4l2_ctrl *hflip;
 	struct v4l2_ctrl *vblank;
 	struct v4l2_ctrl *hblank;
-	struct v4l2_ctrl *red_balance;
-	struct v4l2_ctrl *blue_balance;
-	struct v4l2_ctrl *notify_gains;
 	struct v4l2_ctrl *hdr_mode;
 	struct v4l2_ctrl *link_freq;
 
@@ -1205,12 +1202,12 @@ static int imx708_set_ctrl(struct v4l2_c
 	case V4L2_CID_NOTIFY_GAINS:
 		ret = imx708_write_reg(imx708, IMX708_REG_COLOUR_BALANCE_BLUE,
 				       IMX708_REG_VALUE_16BIT,
-				       imx708->notify_gains->p_new.p_u32[0]);
+				       ctrl->p_new.p_u32[0]);
 		if (ret)
 			break;
 		ret = imx708_write_reg(imx708, IMX708_REG_COLOUR_BALANCE_RED,
 				       IMX708_REG_VALUE_16BIT,
-				       imx708->notify_gains->p_new.p_u32[3]);
+				       ctrl->p_new.p_u32[3]);
 		break;
 	case V4L2_CID_WIDE_DYNAMIC_RANGE:
 		/* Already handled above. */
@@ -1842,8 +1839,7 @@ static int imx708_init_controls(struct i
 		/* The "Solid color" pattern is white by default */
 	}
 
-	imx708->notify_gains = v4l2_ctrl_new_custom(ctrl_hdlr,
-						    &imx708_notify_gains_ctrl, NULL);
+	v4l2_ctrl_new_custom(ctrl_hdlr, &imx708_notify_gains_ctrl, NULL);
 
 	imx708->hdr_mode = v4l2_ctrl_new_std(ctrl_hdlr, &imx708_ctrl_ops,
 					     V4L2_CID_WIDE_DYNAMIC_RANGE,
